spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,223 @@
+          if (!(type.equals(Schema.Type.INT) || type.equals(Schema.Type.DOUBLE) || type.equals(Schema.Type.FLOAT) ||
+            type.equals(Schema.Type.LONG))) {
+            return structuredRecord.get(field);
+          } else {
+            return null;
+          }
+        }
+      }).distinct().zipWithIndex().collectAsMap();
+
+      if (!(map == null ||
+        (map.size() == 1 && map.keySet().contains(null)))) {
+        categoricalFeaturesMap.put(field, map);
+      }
+    }
+
+    // load the model from a file in the model fileset
+    JavaSparkContext javaSparkContext = context.getSparkContext();
+    SparkContext sparkContext = JavaSparkContext.toSparkContext(javaSparkContext);
+    final DecisionTreeModel loadedModel = DecisionTreeModel.load(sparkContext, modelLocation.toURI().getPath());
+
+    JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(StructuredRecord record) throws Exception {
+        List<Double> featureList = new ArrayList<Double>();
+        for (String field : fields) {
+          if (categoricalFeaturesMap.keySet().contains(field)) {
+            featureList.add((categoricalFeaturesMap.get(field).get(record.get(field))).doubleValue());
+          } else {
+            featureList.add(new Double(record.get(field).toString()));
+          }
+        }
+
+        Vector vector = Vectors.dense(Doubles.toArray(featureList));
+        double prediction = loadedModel.predict(vector);
+
+        return cloneRecord(record)
+          .set(config.predictionField, prediction)
+          .build();
+      }
+    });
+    return output;
+  }
+
+  /**
+  *Creates a builder based off the given record.
+  */
+  private StructuredRecord.Builder cloneRecord(StructuredRecord record) {
+    Schema schemaToUse = outputSchema != null ? outputSchema : getOutputSchema(record.getSchema());
+    List<Schema.Field> fields = new ArrayList<>();
+    fields.addAll(schemaToUse.getFields());
+    fields.addAll(Arrays.asList(Schema.Field.of(config.predictionField, Schema.of(Schema.Type.DOUBLE))));
+    schemaToUse = Schema.recordOf("records", fields);
+    StructuredRecord.Builder builder = StructuredRecord.builder(schemaToUse);
+    for (Schema.Field field : schemaToUse.getFields()) {
+      if (!config.predictionField.equals(field.getName())) {
+        builder.set(field.getName(), record.get(field.getName()));
+      }
+    }
+    return builder;
+  }
+
+  private Schema getOutputSchema(Schema inputSchema) {
+    return getOutputSchema(inputSchema, config.predictionField);
+  }
+
+  private Schema getOutputSchema(Schema inputSchema, String predictionField) {
+    List<Schema.Field> fields = new ArrayList<>(inputSchema.getFields());
+    fields.add(Schema.Field.of(predictionField, Schema.of(Schema.Type.DOUBLE)));
+    return Schema.recordOf(inputSchema.getRecordName() + ".predicted", fields);
```

Comment:
Not getting any error for the period in schema name while testing.
Please let me know if this still needs to be changed.